engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbDecisionState.java

```diff
@@ -25,55 +26,65 @@
 
 public final class DbDecisionState implements MutableDecisionState {
 
-  private final DbLong dbDecisionKey = new DbLong();
-  private final PersistedDecision dbPersistedDecision = new PersistedDecision();
-  private final DbString dbDecisionId = new DbString();
-
-  private final DbLong dbDecisionRequirementsKey = new DbLong();
-  private final PersistedDecisionRequirements dbPersistedDecisionRequirements =
-      new PersistedDecisionRequirements();
-  private final DbString dbDecisionRequirementsId = new DbString();
-
-  private final DbCompositeKey<DbLong, DbLong> dbDecisionRequirementsKeyAndDecisionKey =
-      new DbCompositeKey<>(dbDecisionRequirementsKey, dbDecisionKey);
-
+  private final DbLong dbDecisionKey;
```

Comment:
:question: Can this be turned into t local variable only?

Reply:
No, it's used all over to access `ZbColumnFamilies.DMN_DECISIONS` directly.

Reply:
I would like to push back a little bit on this. I just started to rewrite the code and it went like this:
```
   private final ColumnFamily<DbForeignKey<DbLong>, PersistedDecision> decisionsByKey;
  ...
  public DbDecisionState(
      final ZeebeDb<ZbColumnFamilies> zeebeDb, final TransactionContext transactionContext) {
    final var dbDecisionKey = new DbLong();
    fkDecision = new DbForeignKey<>(dbDecisionKey, ZbColumnFamilies.DMN_DECISIONS);

    dbPersistedDecision = new PersistedDecision();
    decisionsByKey =
        zeebeDb.createColumnFamily(
            ZbColumnFamilies.DMN_DECISIONS, transactionContext, fkDecision, dbPersistedDecision);

    dbDecisionId = new DbString();
    latestDecisionKeysByDecisionId =
        zeebeDb.createColumnFamily(
            ZbColumnFamilies.DMN_LATEST_DECISION_BY_ID,
            transactionContext,
            dbDecisionId,
            fkDecision);

   ...

  private Optional<PersistedDecision> findDecisionByKey(final long decisionKey) {
    fkDecision.inner().wrapLong(decisionKey);
    return Optional.ofNullable(decisionsByKey.get(fkDecision)).map(PersistedDecision::copy);
  }
```

So basically, in my mind, the goal should be to use the raw key only to construct the foreign key and then use the foreign key throughout. Maybe I missed something. In that case, please let me know.

Reply:
you're right